packages/cardano-graphql-services/test/TxSubmit/TxSubmitHttpServer.test.ts

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Cardano, TxSubmitProvider } from '@cardano-sdk/core';
+import { TxSubmitHttpServer, TxSubmitHttpServerConfig } from '../../src';
+import { getPort } from 'get-port-please';
+import cbor from 'cbor';
+import got from 'got';
+
+const tx = cbor.encode('#####');
+const BAD_REQUEST_STRING = 'Response code 400 (Bad Request)';
+
+describe('TxSubmitHttpServer', () => {
+  let txSubmitProvider: TxSubmitProvider;
+  let txSubmitHttpServer: TxSubmitHttpServer;
+  let port: number;
+  let apiUrlBase: string;
+  let config: TxSubmitHttpServerConfig;
+
+  beforeAll(async () => {
+    port = await getPort();
+    apiUrlBase = `http://localhost:${port}`;
+    config = { listen: { port } };
+    txSubmitProvider = { healthCheck: jest.fn(() => Promise.resolve({ ok: true })), submitTx: jest.fn() };
+  });
+
+  afterEach(async () => {
+    jest.resetAllMocks();
+  });
+
+  describe('healthy and successful submission', () => {
+    beforeAll(async () => {
+      txSubmitHttpServer = TxSubmitHttpServer.create(config, { txSubmitProvider });
+      await txSubmitHttpServer.initialize();
+      await txSubmitHttpServer.start();
+    });
+
+    afterAll(async () => {
+      await txSubmitHttpServer.shutdown();
+    });
+
+    describe('/health', () => {
+      it('forwards the txSubmitProvider health response', async () => {
+        const res = await got(`${apiUrlBase}/health`, {
+          // eslint-disable-next-line sonarjs/no-duplicate-string
+          headers: { 'Content-Type': 'application/json' }
+        });
+        expect(res.statusCode).toBe(200);
+        expect(JSON.parse(res.body)).toEqual({ ok: true });
+      });
+    });
+
+    describe('/submit', () => {
+      it('returns a 200 coded response with a well formed HTTP request', async () => {
+        expect(
+          (
+            await got.post(`${apiUrlBase}/submit`, {
+              body: tx,
+              headers: { 'Content-Type': 'application/cbor' },
+              method: 'post'
+            })
+          ).statusCode
+        ).toEqual(200);
+        expect(txSubmitProvider.submitTx).toHaveBeenCalledTimes(1);
+      });
+
+      it('returns a 400 coded response if the wrong content type header is used', async () => {
+        try {
+          await got.post(`${apiUrlBase}/submit`, {
+            body: tx,
+            headers: { 'Content-Type': 'application/json' }
+          });
+          throw new Error('fail');
+        } catch (error: any) {
+          expect(error.response.statusCode).toBe(400);
+          expect(error.message).toBe(BAD_REQUEST_STRING);
+          expect(txSubmitProvider.submitTx).toHaveBeenCalledTimes(0);
+        }
+      });
+
+      it('returns a 400 coded response if the tx is not sent as binary data', async () => {
+        try {
+          await got.post(`${apiUrlBase}/submit`, {
+            body: '',
+            headers: { 'Content-Type': 'application/json' }
```

Comment:
Header is checked before payload so this test is the same as above, you want to change header to cbor

Reply:
c85267c9ecc7d97854adb8a409021749a9c3f1dd